library/std/src/sys/sgx/thread.rs

```diff
@@ -94,7 +95,18 @@ pub mod wait_notify {
     }
 
     pub fn new() -> (Notifier, Waiter) {
-        let inner = Arc::new(Parker::new_internal());
+        // Safety:
+        // Some other platforms (looking at you, UNIX!) require that the thread
+        // parker is constructed in-place. This is just a noisy way of writing:
+        // ```rust
+        // let parker = Parker::new();
+        // ```
+        let parker = unsafe {
+            let mut place = MaybeUninit::uninit();
+            Parker::new(place.as_mut_ptr());
+            place.assume_init()
+        };
+        let inner = Arc::new(parker);
```

Comment:
Maybe it's better to keep a separate new method for this? If the `thread_parking::id::Parker` ever changes to rely on its location, this unsafe code here would silently become an issue.

Maybe rename the existing `new` functions to `new_in_place`, and add a regular `new() -> Self` function to `thread_parking::id::Parker`?

Reply:
Sounds good!